baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1563 @@
+			err := bmoapis.AddToScheme(scheme)
+
+			if err != nil {
+				log.Printf("AddToScheme failed: %v", err)
+			}
+
+			c := fakeclient.NewFakeClientWithScheme(scheme, &tc.Host)
+
+			actuator, err := newMachineManager(c, nil, nil, &machine, bmmconfig)
+			if err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+
+			// run the function
+			err = actuator.setHostSpec(context.TODO(), &tc.Host)
+			if err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+
+			// get the saved result
+			savedHost := bmh.BareMetalHost{}
+			err = c.Get(context.TODO(), client.ObjectKey{Name: tc.Host.Name, Namespace: tc.Host.Namespace}, &savedHost)
+			if err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+
+			// validate the result
+			if savedHost.Spec.ConsumerRef == nil {
+				t.Errorf("ConsumerRef not set")
+				return
+			}
+			if savedHost.Spec.ConsumerRef.Name != machine.Name {
+				t.Errorf("found consumer ref %v", savedHost.Spec.ConsumerRef)
+			}
+			if savedHost.Spec.ConsumerRef.Namespace != machine.Namespace {
+				t.Errorf("found consumer ref %v", savedHost.Spec.ConsumerRef)
+			}
+			if savedHost.Spec.ConsumerRef.Kind != "Machine" {
+				t.Errorf("found consumer ref %v", savedHost.Spec.ConsumerRef)
+			}
+			if savedHost.Spec.Online != true {
+				t.Errorf("host not set to Online")
+			}
+			if tc.ExpectedImage == nil {
+				if savedHost.Spec.Image != nil {
+					t.Errorf("Expected image %v but got %v", tc.ExpectedImage, savedHost.Spec.Image)
+					return
+				}
+			} else {
+				if *(savedHost.Spec.Image) != *(tc.ExpectedImage) {
+					t.Errorf("Expected image %v but got %v", tc.ExpectedImage, savedHost.Spec.Image)
+					return
+				}
+			}
+			if tc.ExpectUserData {
+				if savedHost.Spec.UserData == nil {
+					t.Errorf("UserData not set")
+					return
+				}
+				if savedHost.Spec.UserData.Namespace != tc.ExpectedUserDataNamespace {
+					t.Errorf("expected Userdata.Namespace %s, got %s", tc.ExpectedUserDataNamespace, savedHost.Spec.UserData.Namespace)
+				}
+				if savedHost.Spec.UserData.Name != testUserDataSecretName {
+					t.Errorf("expected Userdata.Name %s, got %s", testUserDataSecretName, savedHost.Spec.UserData.Name)
+				}
+			} else {
+				if savedHost.Spec.UserData != nil {
+					t.Errorf("did not expect user data, got %v", savedHost.Spec.UserData)
+				}
+			}
+		})
+	}
+}
+
+func TestExists(t *testing.T) {
+	scheme := runtime.NewScheme()
+	err := bmoapis.AddToScheme(scheme)
+
+	if err != nil {
+		log.Printf("AddToScheme failed: %v", err)
+	}
+
+	host := bmh.BareMetalHost{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "somehost",
+			Namespace: "myns",
+		},
+	}
+	c := fakeclient.NewFakeClientWithScheme(scheme, &host)
+
+	testCases := map[string]struct {
+		Client      client.Client
+		Machine     capi.Machine
+		FailMessage string
+		Expected    bool
+	}{
+		"Failed to find the existing host": {
+			Client: c,
+			Machine: capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
```

Comment:
The annotations should be on the baremetalmachine

Reply:
done